File: python/sort/selectionsort.py
def SELECTION_SORT(A):
    n = len(A)
    for i in range(n):
        minIndex = i
        for j in range(i, n):
            if A[j] < A[minIndex]:
                minIndex = j
        A[i], A[minIndex] = A[minIndex], A[i]
    return A

File: java/sort/SelectionSort.java
class SelectionSort
{
  public static void selectionSort(int a[])
  {
    for(int i=0;i<a.length;i++)
    {
      for(int j=i;j<a.length;j++)
      {
        int temp_index=i;
        if(a[temp_index]>a[j])
        {
          temp_index=j;
        }
        int t=a[i];
        a[i]=a[temp_index];
        a[temp_index]=t;
      }
    }
  }
  public static void main(String[] args) {
    int a[]={5,4,2,3,1};
    selectionSort(a);
    for(int i=0;i<a.length;i++)
    {
      System.out.println(a[i]);
    }
  }
}
